File: algorithms/0122.best_time_to_buy_and_sell_stock_ii/best_time_to_buy_and_sell_stock_ii.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut last = 0;
        let mut sum = 0;
        for i in 1..prices.len() {
            let diff = prices[i] - prices[last];
            last = i;
            if diff > 0 {
                // last = i;
                sum += diff;
            }
        }
        sum
    }
}

File: algorithms/0122.best_time_to_buy_and_sell_stock_ii/best_time_to_buy_and_sell_stock_ii.cpp
class Solution {
public:
    int maxProfit(const vector<int>& prices) {
        int profit = 0;
        int last = prices[0];
        for (int i = 1; i < prices.size(); ++i) {
            const int diff = prices[i] - last;
            profit += max(diff, 0);
            last = prices[i];
        }
        return profit;
    }
};